crates/next-core/js/src/entry/next-hydrate.tsx

```diff
@@ -68,3 +76,51 @@ async function loadPageChunk(assetPrefix: string, chunkPath: string) {
 
   console.debug("The page has been hydrated");
 })().catch((err) => console.error(err));
+
+/**
+ * Subscribes to page data updates from the HMR server.
+ *
+ * TODO(alexkirsz): Handle assetPrefix/basePath.
+ */
+function subscribePageData({ assetPrefix }: { assetPrefix: string }) {
+  const pageChunkPath = location.pathname.slice(1);
+  const dataPath = `_next/data/development/${pageChunkPath}.json`;
```

Comment:
I think we also need to update this on client side navigation

Reply:
Good catch. I've updated it so it unsubscribes and resubscribes when navigating to a different page.